### ISSUE TITLE
speed up randomization-based Wilcoxon test statistics 

### ISSUE THREAD

**Issue by pneuvial** [pneuvial/sanssouci#10]
Randomization-based JER calibration for generic test statistics (achieved by passing a testing function) implementer in the branch  generic-randomization-tests is slow. This is because one is doing m*B calls to the test function. 

It is important to keep the current version for genericity and reproducibility, but we should implement a faster version for the specific case of Wilcoxon test statistics (as already done for Welch test statistics), as follows:

* the test statistic is fairly simple, and in fact a version is already implemented in the sansSouci::wilcoxStat function. Caveat: need to check if the definition of the test stat used is the right one (ie the one in `wilcox.test(...)$statistic` as there are several accepted definitions depending on whether the minimal sum of ranks that can be achieved is subtracted or not.

* for the p-values: in order to reproduce the output of `wilcox.test(...)$p.value`, one needs to implement the approximate (normal) p-value implemented in `stats:::wilcox.test.default`. It requires the test statistics, the number of ties, and the group sizes. The relevant code in the above function is:

```
            z <- STATISTIC - n.x * n.y/2
            SIGMA <- sqrt((n.x * n.y/12) * ((n.x + n.y + 1) -
                sum(NTIES^3 - NTIES)/((n.x + n.y) * (n.x + n.y -
                  1))))
            if (correct) {
                CORRECTION <- switch(alternative, two.sided = sign(z) *
                  0.5, greater = 0.5, less = -0.5)
                METHOD <- paste(METHOD, "with continuity correction")
            }
            z <- (z - CORRECTION)/SIGMA
            PVAL <- switch(alternative, less = pnorm(z), greater = pnorm(z,
                lower.tail = FALSE), two.sided = 2 * min(pnorm(z),
                pnorm(z, lower.tail = FALSE)))
```

For the record: `wicox.test` calculates an exact p-value using  `pwilcox` only if there are no ties and less than 50 items per group. 

Once this is done I also suggest:

* dropping the parallelization (which I believe should not be buried in a low-level function)
* dropping the tidy statements (which should not be useful anymore and require the package to depend on the entire tidyverse)

**Comment by pneuvial** [comment]
Done in c6e9b8e4890f3a720b71e065b1ab4b198fca0b1b.
